.NET/Microsoft.Recognizers.Text.DateTime/Extractors/BaseDurationExtractor.cs

```diff
@@ -335,10 +326,35 @@ private List<ExtractResult> MergeMultipleDuration(string text, List<ExtractResul
                     ret.Add(extractorResults[firstExtractionIndex]);
                 }
 
+                // Add list of separate extractions to separaRet, so that there is a 1 to 1 correspondence
+                // between ret (list of merged extractions) and separaRet (list of unmerged extractions)
+                separaRet.Add(separaList);
+
                 firstExtractionIndex = secondExtractionIndex;
             }
 
-            return ret;
+            // If the first and last elements of a group of contiguous extractions are both preceded/followed by modifiers,
+            // they should not be merged, e.g. "last 2 weeks and 3 days ago"
+            List<ExtractResult> newRet = new List<ExtractResult>();

```

Comment:
Do we really need so many lists? Can't we re-use some, or avoid creating others in certain cases?
There is already too much GC pressure.

Reply:
ah maybe instead of newRet, one could insert the unmerged results from separaRet directly into ret replacing the merged ones. I will check.

Reply:
removed newRet